### ISSUE TITLE
High offset updated for a specific consumer group

### ISSUE THREAD

**Issue by smichaut** [edenhill/kafkacat#193]
Hi,
I am not sure this is an issue, rather a lack of understanding from my part...
I am using kafkacat to produce some messages and I have a consumer written in C that consumes them with a group.id... it is a simple 'toy' project with one broker and 3 partitions...
I also use kakfatool to monitor offset and messages in topics. So far so good.
But since I have started to consume messages with a given group.id if I produce some new messages, I see in kafkatool that the high offset value ('End') is updated for this specific group.id!

Therefore if I try to consume these message using another group.id, I see the new grp in kafkatool 'Consumers' part but the messages aren't proceed since the high and offset value for this new grp.id are both equals to the high value of the old one (I see no lag!).
Here is the command I use to produce messages:
cat <file> | kafkacat -b 127.0.0.1:9092 -P -t <my topic> -K:
It seem's there is a 'hard link' between the producer and the group.id (which seems dubious from what I understand from Kafka) but I don't understand why...

Any idea of what I am getting wrong ?

**Comment by edenhill** [comment]
There is no correlation between a producer and a group.id, group.id are only used by the consumer.
If you see zero lag when producing new messages it means you have a consumer for the given group.id that consumes new messages and commits their offsets.

**Comment by edenhill** [comment]
Please provide your full command lines.

**Comment by smichaut** [comment]
Ho I didn't see after publishing my first message,but the &lt; and &gt; chars have been removed...
So the command is
cat myFile.txt | kafkacat -b 127.0.0.1:9092 -P -t myTopic -K:
(in the file I use a key at the beginning of each line to group all the file on the same partition). 
In order to see what puzzles me you can use this command to produce some message and use the following to consume them:
./kafka-console-consumer.sh --bootstrap-server localhost:9092 --topic myTopic --group myGrp1
Using kafkatool you will see, in the Consumers part in 'myGrp1' that End = Offset and Lag=0. Suppose now you produce once again some message with the same command as before: the value of End and Lag has changed accordingly. Good. But now if you run the kafka console consumer command with --group myGrp2 you see that another item is created under Consumers in kafkatool, both Offset and End are equals and no message is read. You have to produce messages a third time a run the console consumer command with --group myGrp2 to start consumming message from that new group.id! This behavior seems strange to me... I think I miss something in my understanding of how Kafka works, I probably use it the wrong way but I can figure out why. And what I see in kafkatool lure me.

To sum up, I observe that if I want to start consuming message with another group.id it comes in 2-fold: first I must create the consumer group and second consume the messages I will now produce!
I hope this is clearer.

**Comment by edenhill** [comment]
Please see the documentation of auto.offset.reset for the Kafka Consumer here:
http://kafka.apache.org/documentation.html#consumerconfigs
